### PR TITLE
Submission 4

### PR DESCRIPTION
**Pull request by alexcombessie**
- Use the real train set instead of the subset
- Recursive Feature Elimination

### CODE REVIEW

**Comment by kwinkunks** [comment]
**0.598**. NIce — great effort, Alex. Definitely the right direction. I am still validating it, so it's possible the score will move slightly up or down a bit. Stay tuned.

**Comment by alexcombessie** [comment]
Thanks!

**Comment by kwinkunks** [comment]
Hey @alexcombessie I have not been able to reproduce this yet... I am not that familiar with `RFECV` — is it possible to access the optimal features, without having to retrain completely, doing the CV again? Seems like if you could change the second `fit` (under **Apply to test**), it would be a lot faster, and then I could implement the random seed thing we're discussing over in #114 .

(FWIW, everything seems to run OK for me, then I Seem to get into an infinite loop, with a lot of these errors:

    /Users/matt/anaconda/envs/python3/lib/python3.5/site-packages/sklearn/feature_selection/rfe.py:198: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
    support_[features[ranks][:threshold]] = False

**Comment by alexcombessie** [comment]
Hey @kwinkunks, Thanks for your message. It seems that `RFECV` does not play well with cross validation functions from `sklearn`. I think the simplest way to do cross-validation would be to use the feature mask from my RFECV as a fixed feature selection method (`df = df[mask]`) before `predict`.  I am working on a new submission that will do something like this, in order to do a `GridSearch` on the hyperparameters of my classifiers.
